bigquery_etl/cli/generate.py

```diff
@@ -99,7 +99,22 @@ def generate_all(ctx, output_dir, target_project, ignore, use_cloud_function):
     """Run all SQL generators."""
     click.echo(f"Generating SQL content in {output_dir}.")
     click.echo(ROOT / SQL_GENERATORS_DIR)
-    for _, cmd in reversed(generate.commands.items()):
+
+    def generator_command_sort_key(command):
+        match command.name:
+            # Run `stable_views` after `glean_usage` because both update `dataset_metadata.yaml` files
+            # and we want the `stable_views` updates to take precedence.
+            case "glean_usage":
+                return (1, command.name)
+            case "stable_views":
+                return (2, command.name)
+            # Run `derived_view_schemas` last in case other SQL generators create derived views.
+            case "derived_view_schemas":
+                return (4, command.name)
+            case _:
+                return (3, command.name)
+
+    for cmd in sorted(generate.commands.values(), key=generator_command_sort_key):
```

Comment:
I'm hoping running the SQL generators in a deterministic order will also reduce the seemingly random unrelated diffs that sometimes get reported in `sql.diff`.  Currently the SQL generators are run in the reverse order that `Path.iterdir()` returns the `sql_generators/*` subdirectories, and according to the [docs](https://docs.python.org/3/library/pathlib.html#pathlib.Path.iterdir) that order is "arbitrary".